_posts/2020-07-06-hhvm-4.65.markdown

```diff
@@ -0,0 +1,44 @@
+---
+title: "HHVM 4.65"
+layout: post
+author: fred
+category: blog
+---
+
+HHVM 4.65 is released! This release marks the end of support for 4.58;
+HHVM 4.59&ndash;4.64 remain supported, as do the 4.32
+and 4.56 LTS releases.
+
+# Highlights
+
+- `is` and `as` checks for reified enums now only check the underlying type,
+  not the values; this makes them consistent with parameter and return type
+  enforcement.
+- added `_with_error()` variants of preg functions, e.g.
+  `preg_grep_with_error()`, `preg_match_with_error()`,
+  `preg_replace_with_error()`, to avoid the need for `preg_last_error()` calls
+  and the global state that entails.
+- added `HH\is_vec_or_varray()` and `HH\is_dict_or_darray()` functions.
+- clearer error messages when incompatible kinds `extend` each other - for
+  example, a class can not `extend` an interface, but it can `implement` one.
+
+# Breaking Changes
+
+- `array()` literals are no longer supported by the parser; `[]` literals were
+  already banned. Replace with `darray[]` or `varray[]`. On older versions of
+  HHVM, `hhast-migrate --php-arrays` can be used to automate this.
```

Comment:
This is news to me, albeit very good news :)

Have we changed the default parsing options, or is the parsing logic entirely removed? Could you point me at the diff?

Reply:
I think the code is still there but turned off by default everywhere.

- runtime: https://github.com/facebook/hhvm/commit/2f549ff4a4b3d0fa4c27a438351edc556dd7efe1
- parser: https://github.com/facebook/hhvm/commit/38a1920c6826f065584ee5fbf121fe22811a3624

Reply:
I wasn't able to figure out a way to do this through hhconfig; I didn't investigate spend much time on it as AIUI Daniel is planning to actually delete the code this week